libherokubuildpack/src/inventory/unit.rs

```diff
@@ -0,0 +1,11 @@
+use crate::inventory::checksum::Digest;
+
+impl Digest for () {
```

Comment:
As far as I can tell, this is unused. Should we remove it?

If we want to keep it - maybe we should give it a name? `pub struct NoChecksum`, `UnknownChecksum`, `AnyDigest`, or something.

Reply:
This is useful in tests where you might not want to have an actual checksum or when you want to read an inventory file where the reading code doesn't have an implementation of the used digest. This will happen when we have a generic CLI tool to work with inventory files.

 Using `()`/unit seems fine to me (unsurprisingly, since I wrote the code in question). There is no need to introduce a custom type to convey "no information", it's actually the formal definition of unit.

Quoting Wikipedia:
> In the area of [mathematical logic](https://en.wikipedia.org/wiki/Mathematical_logic) and [computer science](https://en.wikipedia.org/wiki/Computer_science) known as [type theory](https://en.wikipedia.org/wiki/Type_theory), a unit type is a [type](https://en.wikipedia.org/wiki/Data_type) that allows only one value (and thus can hold no information).

Note: There is a `BogusDigest` used in tests for the inventory. It's different from this as we there actually test that digests are validated and only match if the digest implementation matches as well.

I would say lets keep it around as-is for now, maybe create an issue to re-evaluate later. I'm pretty sure we'll need it again when the CLI is re-introduced (it intentionally wasn't ported with this PR). It's sitting in its own dedicated module, isn't explicitly interwoven with code outside of the module (since it's just a trait implementation). The complexity cost on the code-base is pretty much zero.

Reply:
> This is useful in tests where you might not want to have an actual checksum or when you want to read an inventory file where the reading code doesn't have an implementation of the used digest. This will happen when we have a generic CLI tool to work with inventory files.

There's a lot more information in this explanation than "No Information". Quoting the Rust docs:

> The () type has exactly one value `()`, and is used when there is no other meaningful value that could be returned.

A `NoChecksum` unit could be a more meaningful value to me.

In any case, this isn't blocking for me.